_posts/2018-01-30-quicli.md

```diff
@@ -0,0 +1,294 @@
+that you can use when you are just starting out with Rust
+as well as when you just want to write a quick tool.
+Sadly, like so often,
+I didn't have the time or concentration
+to sit down and really do this.
+And here's the good news:
+For some reason[^9e-talk],
+after all these months,
+I've now published a very early version.
+
+[cli-tips]: {% post_url 2017-08-31-rust-cli-tips %}
+
+[^9e-talk]: Okay, I admit, I promised to give a talk about Rust, and I wanted to present some nice, clean example code. (Of course, in the end I didn't show quicli in the talk.)
+
+I was actually Sunday evening
+when I published the code,
+and I did so without thinking much about it.
+But then something very interesting happened the very next day:
+After a link to quicli's repo was posted to reddit,
+Garrett Berg ([vitiral]) [replied] saying that
+their [stdcli] project was very similar to my endeavor.
+This was was unexpected.
+And not only were out projects similar,
+but we were both willing to join forces
+and try to make one great crate
+that takes the best of both approaches.
+Garrett opened a few issues on [quicli's issue tracker][issues] (thanks!),
+which basically concluded in [#19],
+which was originally titled
+"merge stdcli and this lib".
+
+[vitiral]: https://github.com/vitiral
+[replied]: https://www.reddit.com/r/rust/comments/7s3zsd/quicli_quickly_build_cool_cli_apps_in_rust/dt28i2b/
+[stdcli]: https://github.com/vitiral/stdcli
+[issues]: https://github.com/killercup/quicli/issues
+[#19]: https://github.com/killercup/quicli/issues/19
+
+That sounded pretty awesome and made me really proud:
+Did my library
+– which was barely a day old at this point –
+look cool enough to compete with their existing effort?
+
+But, before blindly agreeing to this,
+I needed to take a step back
+and evaluate what I want quicli to be.
+Because, let's be honest,
+I hadn't really thought about a 'grand quicli vision'
+before.
+And why should I?
+I just wanted a small framework-like thing
+that made writing CLI apps less of a pain.
+But how exactly do I want to to this?
+Thanks to Garrett's questions and comments
+I now have an answer to this.
+
+## What do I want to quicli to be
+
+(The following are basically quotes from [#19].)
+
+My goal is to provide
+an opinionated set
+of convenience functions
+to quickly build CLI apps.
+
+In its implementation,
+you get what I'd call
+"a small framework around your main function":
+quicli's `main!` macro set up some basic things,
+and its prelude gives a few tools
+that are commonly required in CLI apps.
+These tools are either re-exports from existing Rust crates,
+or abstractions around library functions that are more ergonomic
+that better fit the CLI use-case
+(as well as my ergonomic requirements).
+
+What I want you to end up with is code that is
+concise,
+well-structured and boilerplate-free,
+and production-ready.
+Okay, that's nothing new; we all want that.
+But what does this even mean for a CLI app?
+
+- concise
```

Comment:
Since it's the title of a section I'd expect it to be something like

## Concise